README.md
<p align="center"><img src="screenshots/screencast.gif" width=610></p>

<h2 align="center">Pop-up Dictionary for Anki</h2>

<p align="center">
<a title="Latest (pre-)release" href="https://github.com/glutanimate/popup-dictionary/releases"><img src ="https://img.shields.io/github/release-pre/glutanimate/popup-dictionary.svg?colorB=brightgreen"></a>
<a title="License: GNU AGPLv3" href="https://github.com/glutanimate/popup-dictionary/blob/master/LICENSE"><img  src="https://img.shields.io/badge/license-GNU AGPLv3-green.svg"></a>
<a title="Rate on AnkiWeb" href="https://ankiweb.net/shared/info/153625306"><img src="https://glutanimate.com/logos/ankiweb-rate.svg"></a>
<br>
<a title="Buy me a coffee :)" href="https://ko-fi.com/X8X0L4YV"><img src="https://img.shields.io/badge/ko--fi-contribute-%23579ebd.svg"></a>
<a title="Support me on Patreon :D" href="https://www.patreon.com/bePatron?u=7522179"><img src="https://img.shields.io/badge/patreon-support-%23f96854.svg"></a>
<a title="Follow me on Twitter" href="https://twitter.com/intent/user?screen_name=glutanimate"><img src="https://img.shields.io/twitter/follow/glutanimate.svg"></a>
</p>

> Connecting the flashcard-dots

This is an add-on for the spaced-repetition flashcard app [Anki](https://apps.ankiweb.net/). It provides the ability to quickly draw up related facts on words or phrases, just by double-clicking on them.

### Table of Contents <!-- omit in toc -->

<!-- MarkdownTOC levels="1,2,3" -->

- [Installation](#installation)
- [Documentation](#documentation)
- [Building](#building)
- [Contributing](#contributing)
- [License and Credits](#license-and-credits)

<!-- /MarkdownTOC -->

### Installation

#### AnkiWeb <!-- omit in toc -->

The easiest way to install Pop-up Dictionary is through [AnkiWeb](https://ankiweb.net/shared/info/153625306).

#### Manual installation <!-- omit in toc -->


<details>

<summary>Click here to see the instructions</summary>

1. Make sure you have the [latest version](https://apps.ankiweb.net/#download) of Anki 2.1 installed.
2. Download the latest `.ankiaddon` package from the [releases tab](https://github.com/glutanimate/popup-dictionary/releases) (you might need to click on *Assets* below the description to reveal the download links)
3. From Anki's main window, head to *Tools* → *Add-ons*
4. Drag-and-drop the `.ankiaddon` package onto the add-ons list
5. Restart Anki

</details>

### Documentation

For further information on the use of this add-on please check out [the description text](docs/description.md) for AnkiWeb.

### Building

With [Anki add-on builder](https://github.com/glutanimate/anki-addon-builder/) installed:

    git clone https://github.com/glutanimate/popup-dictionary.git
    cd popup-dictionary
    aab build

For more information on the build process please refer to [`aab`'s documentation](https://github.com/glutanimate/anki-addon-builder/#usage).

### Contributing

Contributions are welcome! Please review the [contribution guidelines](./CONTRIBUTING.md) on how to:

- Report issues
- File pull requests
- Support the project as a non-developer

### License and Credits

*Pop-up Dictionary* is *Copyright © 2018-2021 [Aristotelis P.](https://glutanimate.com/) (Glutanimate)*

My work on the initial version of this add-on was partially funded by two fellow Anki users. I would like to thank both of them for their help.

Ships with the following javascript libraries:

- jQuery (v1.12.4), (c) jQuery Foundation, licensed under the MIT license
- qTip2 (v3.0.3), (c) 2011-2018 Craig Michael Thompson, licensed under the MIT license
- jQuery.highlight, (c) 2007-2014 Johann Burkard, licensed under the MIT license
- jQuery Migrate, (c) OpenJS Foundation and other contributors, https://openjsf.org/, licensed under the MIT license

Pop-up Dictionary is free and open-source software. The add-on code that runs within Anki is released under the GNU AGPLv3 license, extended by a number of additional terms. For more information please see the [LICENSE](https://github.com/glutanimate/popup-dictionary/blob/master/LICENSE) file that accompanied this program.

This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY.

----

<b>
<div align="center">The continued development of this add-on is made possible <br>thanks to my <a href="https://www.patreon.com/glutanimate">Patreon</a> and <a href="https://ko-fi.com/X8X0L4YV">Ko-Fi</a> supporters.
<br>You guys rock ❤️ !</div>
</b>

CHANGELOG.md
# Changelog

All notable changes to Pop-up Dictionary will be documented here. You can click on each release number to be directed to a detailed log of all code commits for that particular release. The download links will direct you to the GitHub release page, allowing you to manually install a release if you want.

If you enjoy Pop-up Dictionary, please consider supporting my work on Patreon, or by buying me a cup of coffee :coffee::

<p align="center">
<a href="https://www.patreon.com/glutanimate" rel="nofollow" title="Support me on Patreon 😄"><img src="https://glutanimate.com/logos/patreon_button.svg"></a>      <a href="https://ko-fi.com/X8X0L4YV" rel="nofollow" title="Buy me a coffee 😊"><img src="https://glutanimate.com/logos/kofi_button.svg"></a>
</p>

:heart: My heartfelt thanks goes out to everyone who has supported this add-on through their tips, contributions, or any other means (you know who you are!). All of this would not have been possible without you. Thank you for being awesome!

## [Unreleased]

## [1.0.0-beta.1] - 2021-08-04

### [Download](https://github.com/glutanimate/popup-dictionary/releases/tag/v1.0.0-beta.1)

### Added

- Added suport for modern Anki versions (tested up to Anki 2.1.46)
- The add-on now sports a sleek new pop-up design and effects, optimized both for light mode and dark mode
- Searches can now also be triggered by selecting a phrase and right-clicking
- Added an option to exclude new notes from snippet results (thanks to @zjosua!)

### Fixed

- Changed the default add-on hotkey to "Alt+Shift+D" (⌥+⇧+D on macOS) to avoid a conflict with default key bindings on Anki 2.1.41 and up
- Fixed an issue that would cause an error when opening cards in the browser (thanks to @Nanco300 for the report!)
- Fixed an issue that would cause images within the pop-ups to not be sized correctly (thanks to @padenw24 for the report!)
- Made the add-on more resilient against conflicts with user-made deck styling
- Made the add-on more resilient against conflicts with other add-ons (tested against the top 200 most popular add-ons)

### Changed

- Refactored major parts of the add-on, making it more robust and future-proof
- Dropped support for older Anki versions. The minimum supported version is now 2.1.23.
- Upgraded qTip2 to v3.0.3. This should hopefully fix a number of bugs with the pop-ups and make for an overall smoother user experience (thanks to @ansaso!)

## [1.0.0-dev.1] - 2019-08-27

### [Download](https://github.com/glutanimate/popup-dictionary/releases/tag/v1.0.0-dev.1)

### Changed

- Dropped 2.0.x support
- Refactored major parts of the add-on

## [0.5.0-beta.1] - 2019-02-28

### Changed

- Added footer to tooltip
- Updated packaging scheme to simplify installation process
    
## [0.4.2] - 2018-08-19

### Fixed

- Quick fix for Anki 2.1 support.
    
## [0.4.1] - 2018-03-18

### Changed

- Changed: Renamed add-on to Pop-up Dictionary
- Changed: New default values for excluded fields

### Fixed
    
- Fixed: Anki 2.1 support

## [0.4.0] - 2018-03-06

### Added

- **New**: Anki 2.1 support (please wait for the next 2.1 beta)
- **New**: merged original dictionary lookup functionality with new note snippets
- **New**: highlight search terms
- **New**: show note in browser
- **New**: hotkey to invoke tooltip manually on custom selection (Ctrl+Shift+D)
- **New**: warn when looking up term with too many hits
- **New**: invoke tooltip on double-click rather than select
- **New**: config file with full support of Anki 2.1's config editor

### Fixed

- **Fixed**: a lot of smaller issues and bugs
    
## [0.3.0] - 2018-03-05

### Added

- nested tooltips
- refined theme
    
## [0.2.0] - 2018-03-03

### Added

- Switch to displaying snippets of notes in the current deck, instead of using the dictionary deck

## v0.1.0 - 2018-02-19

### Added

Initial release

[Unreleased]: https://github.com/glutanimate/popup-dictionary/compare/v1.0.0-dev.1...HEAD
[1.0.0-dev.1]: https://github.com/glutanimate/popup-dictionary/compare/v0.5.0-beta.1...v1.0.0-dev.1
[0.5.0-beta.1]: https://github.com/glutanimate/popup-dictionary/compare/v0.4.2...v0.5.0-beta.1
[0.4.2]: https://github.com/glutanimate/popup-dictionary/compare/v0.4.1...v0.4.2
[0.4.1]: https://github.com/glutanimate/popup-dictionary/compare/v0.4.0...v0.4.1
[0.4.0]: https://github.com/glutanimate/popup-dictionary/compare/v0.3.0...v0.4.0
[0.3.0]: https://github.com/glutanimate/popup-dictionary/compare/v0.2.0...v0.3.0
[0.2.0]: https://github.com/glutanimate/popup-dictionary/compare/v0.1.0...v0.2.0

-----

The format of this file is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

src/popup_dictionary/consts.py

# -*- coding: utf-8 -*-

# Pop-up Dictionary Add-on for Anki
#
# Copyright (C)  2018-2021 Aristotelis P. <https://glutanimate.com/>
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU Affero General Public License as
# published by the Free Software Foundation, either version 3 of the
# License, or (at your option) any later version, with the additions
# listed at the end of the license file that accompanied this program.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU Affero General Public License for more details.
#
# You should have received a copy of the GNU Affero General Public License
# along with this program.  If not, see <https://www.gnu.org/licenses/>.
#
# NOTE: This program is subject to certain additional terms pursuant to
# Section 7 of the GNU Affero General Public License.  You should have
# received a copy of these additional terms immediately following the
# terms and conditions of the GNU Affero General Public License that
# accompanied this program.
#
# If not, please request a copy through one of the means of contact
# listed here: <https://glutanimate.com/contact/>.
#
# Any modifications to this file must keep this entire header intact.

"""
Addon-wide constants
"""

from ._version import __version__

try:
    from .data.patrons import MEMBERS_CREDITED, MEMBERS_TOP
except ImportError:
    MEMBERS_CREDITED = MEMBERS_TOP = ()

__all__ = [
    "ADDON"
]

# PROPERTIES DESCRIBING ADDON


class ADDON(object):
    """Class storing general add-on properties
    Property names need to be all-uppercase with no leading underscores
    """
    NAME = "Pop-up Dictionary"
    MODULE = "popup_dictionary"
    ID = "153625306"
    VERSION = __version__
    LICENSE = "GNU AGPLv3"
    AUTHORS = (
        {"name": "Aristotelis P. (Glutanimate)", "years": "2018-2019",
         "contact": "https://glutanimate.com"},
    )
    AUTHOR_MAIL = "ankiglutanimate@gmail.com"
    LIBRARIES = (
        {"name": "qTip2", "version": "v3.0.3",
         "author": "Craig Michael Thompson", "license": "MIT license",
         "url": "http://qtip2.com/"},
        {"name": "jquery-migrate", "version": "3.0.0",
         "author": "jquery", "license": "MIT license",
         "url": "https://github.com/jquery/jquery-migrate"},
        {"name": "jQuery.highlight", "version": "5",
         "author": "Johann Burkard", "license": "MIT license",
         "url": "https://johannburkard.de/blog/programming/javascript/highlight-javascript-text-higlighting-jquery-plugin.html"},
    )
    CONTRIBUTORS = ()
    SPONSORS = ()
    MEMBERS_CREDITED = MEMBERS_CREDITED
    MEMBERS_TOP = MEMBERS_TOP
    LINKS = {
        "patreon": "https://www.patreon.com/glutanimate",
        "bepatron": "https://www.patreon.com/bePatron?u=7522179",
        "coffee": "http://ko-fi.com/glutanimate",
        "description": "https://ankiweb.net/shared/info/{}".format(ID),
        "rate": "https://ankiweb.net/shared/review/{}".format(ID),
        "twitter": "https://twitter.com/glutanimate",
        "youtube": "https://www.youtube.com/c/glutanimate",
        "help": ""
    }
